diff --git a/swig/python/setup.py.in b/swig/python/setup.py.in
index c22d19f113..867b087c4f 100644
--- a/setup.py
+++ b/setup.py
@@ -230,13 +230,13 @@ class gdal_ext(build_ext):
         if self.include_dirs is None:
             self.include_dirs = include_dirs
         # Needed on recent MacOSX
-        elif isinstance(self.include_dirs, str) and sys.platform == 'darwin':
+        elif include_dirs and isinstance(self.include_dirs, str) and sys.platform == 'darwin':
             self.include_dirs += ':' + ':'.join(include_dirs)
 
         if self.library_dirs is None:
             self.library_dirs = library_dirs
         # Needed on recent MacOSX
-        elif isinstance(self.library_dirs, str) and sys.platform == 'darwin':
+        elif library_dirs and isinstance(self.library_dirs, str) and sys.platform == 'darwin':
             self.library_dirs += ':' + ':'.join(library_dirs)
 
         if self.libraries is None:
